lib/active_admin/csv_builder.rb

```diff
@@ -53,11 +53,19 @@ def build(controller, csv)
         csv << CSV.generate_line(columns.map{ |c| encode c.name, options }, csv_options)
       end
 
-      (1..paginated_collection.total_pages).each do |page|
-        paginated_collection(page).each do |resource|
-          resource = controller.send :apply_decorator, resource
-          csv << CSV.generate_line(build_row(resource, columns, options), csv_options)
+      add_collection_to_csv = -> {
+        (1..paginated_collection.total_pages).each do |page|
+          paginated_collection(page).each do |resource|
+            resource = controller.send :apply_decorator, resource
+            csv << CSV.generate_line(build_row(resource, columns, options), csv_options)
+          end
```

Comment:
Could `add_collection_to_csv` be defined as a method instead of a lambda?

Reply:
@varyonic sure thing, updated `add_collection_to_csv` to be a method!